[VO-1227] chore: Migrate to Rsbuild

cballevre opened the pull request:
```
### 🔧 Tech

* Migrate to Rsbuild
```

**Related PRs:**
- https://github.com/cozy/cozy-libs/pull/2628

cballevre committed:
feat: Add Rsbuild dependencies

cballevre committed:
feat: Add Rsbuild config

cballevre committed:
feat: Add Rsbuild scripts

cballevre committed:
refactor: Move targets to src root

cballevre committed:
feat: Add alias to tsconfig

cballevre committed:
feat: Migrate index.ejs from targets

cballevre committed:
feat: Move static assets into the public folder

cballevre committed:
refactor: Move manifest.webapp at root

cballevre committed:
feat: Use import with extension for styles

cballevre committed:
refactor: Use only deep import for cozy-ui

cballevre committed:
feat: Use common alias

cballevre committed:
feat: Remove __APP_SLUG__ to use cozy-photos slug directly

cballevre committed:
feat: Remove webpack custom config

cballevre committed:
feat: Remove cozy-client-js

cballevre committed:
feat: Use flag debug to activate redux devtools

cballevre committed:
feat: Remove mobile target

cballevre committed:
feat: Remove jsconfig.json

cballevre committed:
feat: Every one use assets

cballevre committed:
feat: Remove old package

cballevre committed:
feat: Remove style linting temporary

zatteo commented:
@cballevre very nice work

Ldoppea commented on src/targets/browser/index.jsx:
```diff
@@ -57,17 +55,12 @@ const setupAppContext = memoize(() => {
   // Enable Redux dev tools
   const composeEnhancers =
-    (__DEVELOPMENT__ && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose
+    (flag('debug') && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose
```
As the flag plugin is initialized on client's `login` event and because `client.login()` is asynchronous, it seems like relying on `flag('debug')` here would be subject to race condition, wouldn't it?

I suppose it would work here because on web apps, the `client.login()` only access local resources, so it would be fast enough to ends before the composeEnhancers initialisation. But it seems a bit risky.

Ldoppea commented:
This commit message says the opposite of what is done. Is the error on the message? or on what have been done?
![image](https://github.com/user-attachments/assets/711c68f7-686c-47cf-952c-5f044a6a91d5)

cballevre commented:
This a mistake on the message :slightly_frowning_face:

Ldoppea committed:
feat: Use `development` mode when using `watch` script

In cozy-script we used to enforce the development mode when doing a
`yarn watch`

The development mode changes the build behavior by removing some steps
like minification which results to a faster build

A side effect is that the code generated from the `watch` command is a
bit different than the one generated from `build` command. This may
produce some differences on the app's behavior, but this is rare enough
(happened 1 or 2 times in the past years) so we consider the speed gain
to still be valuable. Also we know that we should run a `build` locally
as ultimate check before pushing new code to the git repo

So we want to enable this mode in the new Rsbuild configuration

Here are the timings for initial build and then 4 differents edits in
the code that trigger a re-build

yarn watch
```
    7,57s (initial build)
    5,79s
    5,83s
    6,68s
    7,15s
```

yarn watch --mode development
```
    7,00s (initial build)
    0,69s
    0,45s
    0,92s
    0,46s
```

Related code:
https://github.com/cozy/create-cozy-app/blob/master/packages/cozy-scripts/scripts/watch.js#L13

Ldoppea committed:
chore: Adapt bundlemon config to new Rsbuild output

Ldoppea committed:
feat: Upgrade rsbuild-config-cozy-app to `0.2.0`

`rsbuild-config-cozy-app` has been upgraded to `0.2.0` in order to
retrieve the configuration needed to run `rsbuild dev` with our
cozy-stack based architecture and then benefits from HMR feature

Related PR: cozy/cozy-libs#2700

Ldoppea committed:
feat: Add `yarn start` script for rsbuild

Since previous commit we can run `rsbuild dev` to enable HMR feature

We now have 3 different scripts:
- `yarn build`: build the app for production
- `yarn watch`: like for the `build` script but with some optimizations
  in order to build faster and to trigger rebuild automatically when
  the code changes
- `yarn dev`: build the app with HMR enabled. This should be the
  fastest way to debug the app, but the HMR feature may prevent the app
  to work in some environments (i.e. flagship app)

Ldoppea committed:
feat: Use SWC for Jest unit tests

Ldoppea committed:
feat: Upgrade rsbuild-config-cozy-app to `0.2.1`

`rsbuild-config-cozy-app` has been upgraded to `0.2.1` in order to
retrieve the configuration needed to run intents and public targets in
the `rsbuild dev` mode

Related PR: cozy/cozy-libs#2717

Ldoppea commented:
Non-regression tests have been done:
- [x] basic app usage
- [x] upload photo
- [x] download photo

____

Performances comparison on my computer:

`yarn build`
    
| | Webpack  | Rsbuild |
| ------------- | ------------- | ------------- |
| Run 1 | 33,91s  | 7,28s  |
| Run 2 | 31,61s  | 6,83s  |
| Run 3 | 30,06s  | 7,49s |
| Run 4 | 33,09s  | 6,39s  |
| Run 5 | 37,45s  | 7,12s  |
| **Average** | **33,22s**  | **7,02s**  |

`yarn watch`

| | Webpack  | Rsbuild |
| ------------- | ------------- | ------------- |
| Initial build | 17,50s  | 7,00s  |
| Rebuild 1 | 2,45s  | 0,69s  |
| Rebuild 2 | 2,65s  | 0,45s |
| Rebuild 3 | 2,33s  | 0,92s  |
| Rebuild 4 | 2,26s  | 0,46s  |
| **Average** (initial build removed from average) | **2,42s**  | **0,63s**  |

zatteo committed:
feat: Add style linting again

zatteo committed:
fix: Inject correctly custom themes override

zatteo committed:
feat: Upgrade rsbuild-config-cozy-app to 0.3.0

`rsbuild-config-cozy-app` has been upgraded to `0.3.0` in order to
retrieve the configuration needed to import images and static assets

Related PR: [cozy/cozy-libs#2728](cozy/cozy-libs#2728)

zatteo commented on src/targets/browser/index.ejs:
```diff
@@ -14,10 +14,14 @@
+  <div role="application" data-cozy="{{.CozyData}}"></div>
+  <% htmlPlugin.files.js.forEach(function(file) { %>
+    <script defer src="<%- file %>"></script>
+  <% }); %>
```
Any idea why there is no <body> here ?
